### PR TITLE
Bind BackgroundImageAlignment to Settings UI

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request
This binds `backgroundImageAlignment` to the Settings UI. A custom control was created for this (see Demo below). The implementation involves...
- storing a reference to all the ToggleButtons
- storing the `ConvergedAlignment` as a tag for each button
- adding the same click handler for each of these buttons, that just clears all the other button states, and activates the current one.

## References
#1564 - Settings UI

## Demo
Background Image Alignment Control

Tested:
- value matches JSON set one
- clicking on selected value keeps the same button checked
- clicking an unselected button works property

### CODE REVIEW

**Comment by carlos-zamora** [comment]
### Notes for Reviewers
This PR is still in a draft state because...
- [ ] Bug: ComboBox does not initialize to correct value on startup
  - I've stepped through but I can't see what's inside the `EnumEntry`
  - Setting to new alignments via the JSON and SUI works fine. And getter for BIAlignment seems to be tested just fine via the `TerminalSettings` call.
- [ ] 3x3-grid toggle button control
  - We want to replace the ComboBox with a 3x3-grid of toggle buttons where only 1 can be `IsChecked` at a time.
  - Need to learn how to...
    - leverage `RadioButtons` but replacing the control with a `ToggleButton`. Need help to do that.
    - fill `ToggleButton`s with [Segoe MDL2 icons](https://docs.microsoft.com/en-us/windows/uwp/design/style/segoe-ui-symbol-font) E744 (rotated for each corner) and E745/E746 (flipped for each edge)

If anybody comes along with suggestions as to how to accomplish this, let me know 😊

File: src/cascadia/TerminalSettingsEditor/Profiles.cpp [hunk 1]
```diff
@@ -153,4 +170,22 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
...
+    {
+        if (auto button{ sender.try_as<Windows::UI::Xaml::Controls::Primitives::ToggleButton>() })
+        {
+            if (auto tag{ button.Tag().try_as<int32_t>() })
+            {
+                // Update the Profile's value
+                _State.Profile().BackgroundImageAlignment(static_cast<ConvergedAlignment>(*tag));
+
+                // reset all of the buttons to unchecked, except for the one that was clicked
+                for (auto biButton : _BIAlignmentButtons)
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/Profiles.cpp:184]: `const &`

**Review by DHowett — APPROVED** [review]
This is exactly what we talked about, and it is great. It's so hard to do anything else.

**Review by miniksa — APPROVED** [review]

**Comment by mdtauk** [comment]
I think with a bit of styling, and possibly some new icons, this image alignment UI could be made a little more elegant.

Here are some ideas...

image

**Comment by mdtauk** [comment]
If these are a bit too large to include, here are some at the smallest size allowed by the Touch Targets.

image
_Each button is 32 x 32, and the whole control is 100 x 100_

**Comment by carlos-zamora** [comment]
Personally, I like the one on the bottom right most (icons aren't in squares and corners are rounded). That looks _super slick_! Curious what @cinnamon-msft and @DHowett think. I'm all in for a redesign, not sure if it warrants its own issue though.

As for how to implement these changes, that shouldn't be too difficult. Since we have each button have its own space in the grid, we would have to apply a [CornerRadius](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.CornerRadius?view=winrt-19041) just to the appropriate corner. And the icons have to be packaged used to replace the existing ones. Looks like we'd just need 3 icons too (corner, edge, and center), because we could just use render transforms to make them look right, like we do now.

**Comment by mdtauk** [comment]
> As for how to implement these changes, that shouldn't be too difficult. Since we have each button have its own space in the grid, we would have to apply a [CornerRadius](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.CornerRadius?view=winrt-19041) just to the appropriate corner. And the icons have to be packaged used to replace the existing ones. Looks like we'd just need 3 icons too (corner, edge, and center), because we could just use render transforms to make them look right, like we do now.
>
It would probably be easier to just include individual icon assets, instead of messing around with transforms and rotations.  Not sure if Terminal packages it's icons as a font, via SVG, or if it exclusively uses **Segoe UI MDL2 Assets**.

I included both MDL2 and Fluent versions for when the transition happens.

The Fluent example uses the Filled and Outlined versions of the icons, which would mean double the assets, which is the intent for active and inactive states.

**Pull request merged**